frontend/packages/dev-console/src/components/EmptyState.tsx

```diff
@@ -2,41 +2,67 @@ import * as React from 'react';
 import { Gallery, GalleryItem } from '@patternfly/react-core';
 import { CatalogTile } from '@patternfly/react-catalog-view-extension';
 import { connect } from 'react-redux';
-import { history, PageHeading } from '@console/internal/components/utils';
-import { formatNamespacedRouteForResource } from '@console/shared/src/utils/namespace';
-import { FLAG_KNATIVE_EVENTING } from '@console/knative-plugin';
-import * as importGitIcon from '../images/from-git.svg';
-import * as yamlIcon from '../images/yaml.svg';
-import * as dockerfileIcon from '../images/dockerfile.svg';
-import { useAddToProjectAccess } from '../utils/useAddToProjectAccess';
-import { allCatalogImageResourceAccess, allImportResourceAccess } from '../actions/add-resources';
+import { history, PageHeading, useAccessReview } from '@console/internal/components/utils';
+import { useExtensions } from '@console/plugin-sdk';
+import { RootState } from '@console/internal/redux';
+import { isAddAction, AddAction } from '../extensions/add-actions';
 import './EmptyState.scss';
 
+const navigateTo = (e: React.SyntheticEvent, url: string) => {
+  history.push(url);
+  e.preventDefault();
+};
+
+interface ItemProps {
+  action: AddAction;
+  namespace: string;
+}
+
+const Item: React.FC<ItemProps> = ({
+  action: {
+    properties: { label, description, icon, iconClass, url, accessReview },
+  },
+  namespace,
+}) => {
+  const access =
+    !accessReview ||
+    (Array.isArray(accessReview) ? accessReview : [accessReview])
```

Comment:
I'd suggest to declare `accessReview` simply as:

```ts
accessReview?: AccessReviewResourceAttributes[];
```

If that's not feasible, we can use `_.castArray` to simplify the coercion to array object:

```ts
_.castArray(accessReview).map( /* .. */ );
```

Reply:
Converted to array property.